lib/experimental/fonts/font-library/class-wp-font-collection.php

```diff
@@ -21,41 +21,128 @@
+	 *      @type array  $categories The font collection's categories.
+	 *  }
 	 */
 	public function __construct( $config ) {
-		if ( empty( $config ) || ! is_array( $config ) ) {
-			throw new Exception( 'Font Collection config options is required as a non-empty array.' );
-		}
+		$this->is_config_valid( $config );
+
+		$this->slug          = isset( $config['slug'] ) ? $config['slug'] : '';
+		$this->name          = isset( $config['name'] ) ? $config['name'] : '';
+		$this->description   = isset( $config['description'] ) ? $config['description'] : '';
+		$this->src           = isset( $config['src'] ) ? $config['src'] : '';
+		$this->font_families = isset( $config['font_families'] ) ? $config['font_families'] : array();
+		$this->categories    = isset( $config['categories'] ) ? $config['categories'] : array();
+	}
 
-		if ( empty( $config['slug'] ) || ! is_string( $config['slug'] ) ) {
-			throw new Exception( 'Font Collection config slug is required as a non-empty string.' );
+	/**
+	 * Checks if the font collection config is valid.
+	 *
+	 * @since 6.5.0
+	 *
+	 * @param array $config Font collection config options. {
+	 *      @type string $slug        The font collection's unique slug.
+	 *      @type string $name        The font collection's name.
+	 *      @type string $description The font collection's description.
+	 *      @type string $src         The font collection's source.
+	 *      @type array  $font_families An array of font families in the font collection.
+	 *      @type array  $categories The font collection's categories.
+	 *  }
+	 * @return bool True if the font collection config is valid and false otherwise.
+	 */
+	public static function is_config_valid( $config ) {
+		if ( empty( $config ) || ! is_array( $config ) ) {
+			_doing_it_wrong( __METHOD__, __( 'Font Collection config options are required as a non-empty array.', 'gutenberg' ), '6.5.0' );
+			return false;
 		}
 
-		if ( empty( $config['name'] ) || ! is_string( $config['name'] ) ) {
-			throw new Exception( 'Font Collection config name is required as a non-empty string.' );
+		$required_keys = array( 'slug', 'name' );
+		foreach ( $required_keys as $key ) {
+			if ( empty( $config[ $key ] ) ) {
+				_doing_it_wrong(
+					__METHOD__,
+					// translators: %s: Font collection config key.
+					sprintf( __( 'Font Collection config %s is required as a non-empty string.', 'gutenberg' ), $key ),
+					'6.5.0'
+				);
+				return false;
+			}
 		}
 
-		if ( ( empty( $config['src'] ) || ! is_string( $config['src'] ) ) && ( empty( $config['data'] ) ) ) {
-			throw new Exception( 'Font Collection config "src" option OR "data" option is required.' );
+		if (
+			( empty( $config['src'] ) && empty( $config['font_families'] ) ) ||
+			( ! empty( $config['src'] ) && ! empty( $config['font_families'] ) )
+		) {
+			_doing_it_wrong( __METHOD__, __( 'Font Collection config "src" option OR "font_families" option are required.', 'gutenberg' ), '6.5.0' );
```

Comment:
Best to use sprintf placeholders for `src` and `font_families` here to prevent accidental translation

Reply:
added here: https://github.com/WordPress/gutenberg/pull/58222/commits/18f996342b4ba37542eaa87a727a409997bd162f